.github/workflows/build.yml

```diff
@@ -290,9 +290,9 @@ jobs:
         run: ls dist/*
 
       - name: Save Artifacts
-        uses: actions/upload-artifact@v2
+        uses: actions/upload-artifact@v4
         with:
-          name: epicscorelibs
+          name: epicscorelibs-${{ matrix.os }}-${{ matrix.python || matrix.pyver }}
```

Comment:
Looks correct in principle.  I think the failures could be resolved by appending something like `${{ matrix.piparch || 'source' }}`.